### ISSUE TITLE
Container host ports show up as "0.0.0.0" in "ports" tab in container detail page and in edit container page.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#7690]
Rancher Versions:v1.4.0-rc9

Steps to reproduce the problem:
Create a container on a host that has no "scheduler ips".
Container host ports show up as "0.0.0.0" in "ports" tab in container detail page and in edit container page.

<img width="1171" alt="screen shot 2017-02-04 at 12 20 23 pm" src="https://cloud.githubusercontent.com/assets/4266958/22621256/5525d396-ead4-11e6-958a-88ad947031b7.png">

<img width="957" alt="screen shot 2017-02-04 at 12 20 06 pm" src="https://cloud.githubusercontent.com/assets/4266958/22621258/59c8938e-ead4-11e6-81fa-4f28e3727788.png">

Container has ports set as:
```
"ports": [
"0.0.0.0:9091:80/tcp"
],
```
Metadata returns 0.0.0.0 for host Ip:

wget http://rancher-metadata/latest/self/container/ports/0 returns ```0.0.0.0:3456:80/tcp```

Not sure which one is being used in this case. Both may need to be addressed.

**Comment by StrongMonkey** [comment]
I believe this is expected. If you don't specified any host ip, by default docker and cni plugin will bind 0.0.0.0 to all its port binding, which means binding to all ip4 address.

**Comment by sangeethah** [comment]
@StrongMonkey , From users perspective, he needs to know the Ip he can use to access the exposed port which in this case is the host's ip on which the container is running.

Also metadata for the container's port , should expose only the hostIp for the ports. This is what I see return for the earlier versions of metadata. Only the latest version of metadata is having this issue of returning "0.0.0.0" instead of host Ip for container ports.

**Comment by vincent99** [comment]
`0.0.0.0` is what it really is, so that's what should be returned. There are an unknown number of IPs that the container might be reachable by. In the UI we can link to the primary host IP so you can click it, but returning it from the API as if that is the actual bound address is just wrong.

**Comment by cjellick** [comment]
@vincent99 I was thinking the UI could be something like:

```
0.0.0.0
(192.168.2.111)
```
or 
```
0.0.0.0 (192.168.2.111)
```
where that IP in parens is the host IP. What do you think?

I'm going to assign to you @vincent99

**Comment by westlywright** [comment]
This will be addressed in 2.0 as we will show the public endpoints to the user (per @vincent99 ) so im going to close this
